student/src/main/java/tds/student/services/remote/RemoteAdaptiveService.java

```diff
@@ -51,14 +51,14 @@ public PageGroup createNextItemGroup(final TestOpportunity testOpportunity, fina
       return pageGroup;
     }
 
-    List<OpportunityItem> items = examItemResponseRepository.createNextItemGroup(testOpportunity.getOppInstance().getExamId(), lastPage, lastPosition);
+    final List<OpportunityItem> items = examItemResponseRepository.createNextItemGroup(testOpportunity.getOppInstance().getExamId(), lastPage, lastPosition);
 
-    PageGroup remotePageGroup = new PageGroup(items.get(0));
+    final PageGroup remotePageGroup = PageGroup.Create(items);
 
-    for (OpportunityItem item : items) {
-      ItemResponse response = new ItemResponse(item);
-      response.setPrefetched(true);
-      remotePageGroup.add(response);
+    if (remotePageGroup != null) {
+      for (final ItemResponse response : remotePageGroup) {
+        response.setPrefetched(true);
+      }
     }
```

Comment:
Added this to handle the end of a test, when there are no additional item groups to be created.  I also changed PageGroup to use the static Create factory method.  (Note that passing the factory an empty collection results in a NULL PageGroup value.)